internal/elasticsearch/pipeline/resource.go

```diff
@@ -0,0 +1,49 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package pipeline
+
+import (
+	"encoding/json"
+	"strings"
+
+	"github.com/pkg/errors"
+	"gopkg.in/yaml.v3"
+)
+
+// Resource represents a pipeline resource loaded from a file
+type Resource struct {
+	Name    string // Name of the pipeline
+	Format  string // Format (extension) of the pipeline
+	Content []byte // Content is the original file contents.
+}
+
+// FileName returns the original filename associated with the pipeline.
+func (p *Resource) FileName() string {
+	pos := strings.LastIndexByte(p.Name, '-')
+	if pos == -1 {
+		pos = len(p.Name)
+	}
+	return p.Name[:pos] + "." + p.Format
+}
+
+// ToJSON returns the pipeline contents in JSON format.
+func (p *Resource) ToJSON() (asJSON []byte, err error) {
+	switch p.Format {
+	case "json":
+		asJSON = p.Content
+	case "yaml", "yml":
+		var node map[string]interface{}
+		err = yaml.Unmarshal(p.Content, &node)
+		if err != nil {
+			return nil, errors.Wrapf(err, "unmarshalling pipeline content failed (pipeline: %s)", p.Name)
+		}
+		if asJSON, err = json.Marshal(&node); err != nil {
```

Comment:
`&` operator is not need for marshaling.

Reply:
done